src/core/service/token/creator.go

```diff
@@ -180,6 +180,9 @@ func (rep repositoryFilter) filter(ctx security.Context, pm promgr.ProjectManage
 	} else if ctx.Can(rbac.ActionPull, resource) {
 		permission = "R"
 	}
+	if ctx.Can(rbac.ActionScannerPull, resource) {
+		permission = fmt.Sprintf("%s%s", permission, "S")
+	}
```

Comment:
Here is very abrupt. What about `ActionScannerPull` as a special action for registry which means `RS` like `RW` for `ActionPush` ? So that it will be like this

```go
	if ctx.Can(rbac.ActionPush, resource) && ctx.Can(rbac.ActionPull, resource) {
		permission = "RWM"
	} else if ctx.Can(rbac.ActionPush, resource) {
		permission = "RW"
	} else if ctx.Can(rbac.ActionScannerPull, resource) {
		permission = "RS"
	} else if ctx.Can(rbac.ActionPull, resource) {
		permission = "R"
	}
```

And maybe we can create a robot which only has (Resource, ScannerPull) policy.

Reply:
agree, updated.